packages/backend/src/services/OAuthService/OAuthService.ts

```diff
@@ -0,0 +1,126 @@
+import {
+    AnyType,
+    AuthTokenPrefix,
+    ForbiddenError,
+    NotFoundError,
+    UserWithOrganizationUuid,
+} from '@lightdash/common';
+import OAuth2Server from '@node-oauth/oauth2-server';
+import { NextFunction, Request, Response } from 'express';
+import { LightdashConfig } from '../../config/parseConfig';
+import { DEFAULT_OAUTH_CLIENT_ID, OAuth2Model } from '../../models/OAuth2Model';
+import { UserModel } from '../../models/UserModel';
+import { BaseService } from '../BaseService';
+
+type OAuthServiceArguments = {
+    userModel: UserModel;
+    oauthModel: OAuth2Model;
+    lightdashConfig: LightdashConfig;
+};
+
+export class OAuthService extends BaseService {
+    protected oauthServer!: OAuth2Server;
+
+    private userModel: UserModel;
+
+    private oauthModel: OAuth2Model;
+
+    private lightdashConfig: LightdashConfig;
+
+    constructor({
+        userModel,
+        oauthModel,
+        lightdashConfig,
+    }: OAuthServiceArguments) {
+        super();
+        this.userModel = userModel;
+        this.oauthModel = oauthModel;
+        this.lightdashConfig = lightdashConfig;
+        this.initializeOAuthServer();
+    }
+
+    private initializeOAuthServer(): void {
+        this.oauthServer = new OAuth2Server({
+            model: this.oauthModel,
```

Comment:
this is how we initialize the oauth2server, the model is a file with some endpoints implemented, for the library to use. https://node-oauthoauth2-server.readthedocs.io/en/master/model/overview.html